### ISSUE TITLE
Query fragment validation fails after set_type_system_hir

### ISSUE THREAD

**Issue by zackangelo** [apollographql/apollo-rs#504]
## Description

When using `set_type_system_hir` to add a pre-compiled type system to an `ApolloCompiler` instance, two seemingly incorrect errors are generated: 

```
Unable to fetch source '': UnknownFileError
Error: cannot find type `Pet` in this document
```

The errors only occur when querying an inline fragment, eg: 
```graphql
... on Dog {
  __typename
  dogBreed
}
```

The errors are not produced at all when compiling the schema and the query on the same `ApolloCompiler` instance. 

All of the underlying compiler data structures that were produced (type definitions, selections, etc) seem correct despite the errors.

### Steps to reproduce

Reproducible example: https://github.com/zackangelo/apollo-testing/tree/fed3a41ba1bc2f8c2164c921916593f181d3fc58

1. Create a compiler 
2. Add a type system to the compiler that contains interfaces
3. Get the type system from the compiler 
4. Create a 2nd compiler
5. Add the type system from the 1st compiler to the 2nd compiler
6. Attempt to execute a query that uses inline fragments

### Expected result

No errors

### Actual result

Error pasted above

### Environment

- Operating system and version: Mac OS X
- Shell (bash/zsh/powershell): zssh
- `apollo-rs` crate: apollo-compiler
- Crate version: 0.7.1

**Comment by lrlna** [comment]
hmmm! that's strange. we use the `set_type_system_hir()` api internally quite a bit, and this is a first. Seems like it has nothing to do with the executable either. just setting the hir again and validating causes this error. I'll investigate!

**Comment by lrlna** [comment]
https://github.com/apollographql/apollo-rs/commit/de4baea13745089ace3821bccc30cf1c4008ba20 should fix it for you! the interfaces query wasn't checking for precomupted hir

**Comment by zackangelo** [comment]
Tested and it works, thanks!

**Comment by zackangelo** [comment]
Resolved by https://github.com/apollographql/apollo-rs/commit/de4baea13745089ace3821bccc30cf1c4008ba20

**Comment by lrlna** [comment]
@zackangelo we published a patch with this fix - `cargo add apollo-compiler@0.7.2`
